### PR TITLE
AV-168302 [Gateway API] Update gateway api docs

### PR DESCRIPTION
**Pull request by aaha97**

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

File: docs/gateway-api/gateway-api-v1beta1.md [hunk 1]
```diff
@@ -82,6 +82,12 @@ A sample Gateway object is shown below:
 
 The above Gateway object would correspond to a single Layer 7 virtual service in the AVI controller, with two ports (80, 443) exposed and a sslKeyAndCertificate created based on the secret **bar-example-com-cert**.
 
+The hostname field `.spec.listeners[i].hostname` is mandatory. It can configured with or without a wildcard, but cannot be only `*`.
```
Comment (pkoshtavmware) [inline docs/gateway-api/gateway-api-v1beta1.md:85]: Might want to change it from "It can configured" to "It can be configured"

**Review by akshayhavile — COMMENTED** [review]
LGTM.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Comment by akshayhavile** [comment]
build ako

**Review by akshayhavile — APPROVED** [review]

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

File: docs/gateway-api/gateway-api-v1beta1.md (hunk 1 above)
Comment (aaha97) [reply docs/gateway-api/gateway-api-v1beta1.md]: fixed, thanks

**Comment by aaha97** [comment]
skip jira-id-check

**Review by pkoshtavmware — APPROVED** [review]
LGTM

File: docs/gateway-api/gateway-api-v1beta1.md [hunk 2]
```diff
@@ -82,6 +82,12 @@ A sample Gateway object is shown below:
 
 The above Gateway object would correspond to a single Layer 7 virtual service in the AVI controller, with two ports (80, 443) exposed and a sslKeyAndCertificate created based on the secret **bar-example-com-cert**.
 
+The hostname field `.spec.listeners[i].hostname` is mandatory. It can be configured with or without a wildcard, but cannot be only `*`.
```
Comment (swathinsankaran) [inline docs/gateway-api/gateway-api-v1beta1.md:85]: All the limitations are under the section `Conditions and Caveats`.

File: docs/gateway-api/gateway-api-v1beta1.md [hunk 3]
```diff
@@ -142,6 +148,12 @@ A sample HTTPRoute object is shown below:
 
 The above HTTPRoute object gets translated to two child VS in the AVI controller. One child VS with match criteria as the path begins with `/bar` and a single Pool Group with a single pool and another child VS with match criteria as path begins with `/foo`, a single Pool Group with two pools, and an HTTP Request policy to add `my-header` to the HTTP request forwarded to the backends.
 
+Hostnames are mandatory and cannot contain wildcard.
```
Comment (swathinsankaran) [inline docs/gateway-api/gateway-api-v1beta1.md:151]: All the limitations are under the section `Conditions and Caveats`.

**Comment by akshayhavile** [comment]
build ako

**Comment by DixitAakash** [comment]
@aaha97 The **values.yaml** related changes for Gateway API are missing from docs which specify/describe fields in values.yaml. Can you please include those.

**Comment by aaha97** [comment]
> @aaha97 The **values.yaml** related changes for Gateway API are missing from docs which specify/describe fields in values.yaml. Can you please include those.

the installation section has the details

File: docs/gateway-api/gateway-api-v1beta1.md (hunk 2 above)
Comment (aaha97) [reply docs/gateway-api/gateway-api-v1beta1.md:85]: it should be fine to have the limitations specified in both the locations

File: docs/gateway-api/gateway-api-v1beta1.md (hunk 3 above)
Comment (aaha97) [reply docs/gateway-api/gateway-api-v1beta1.md:151]: it should be fine to have the limitations specified in both the locations

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Review by DixitAakash — APPROVED** [review]
LGTM

**Comment by DixitAakash** [comment]
build ako

**Review by DixitAakash — APPROVED** [review]
LGTM

**Comment by DixitAakash** [comment]
build ako

**Review by akshayhavile — APPROVED** [review]

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Pull request merged**